plugins/modules/swim_workflow_manager.py

```diff
@@ -1064,119 +1255,158 @@ def get_diff_import(self):
-                family="software_image_management_swim",
-                function=import_function,
-                op_modifies=True,
-                params=import_params,
-            )
-            self.log("Received API response from {0}: {1}".format(import_function, str(response)), "DEBUG")
+                self.log("Received API response from {0}: {1}".format(import_function, str(response)), "DEBUG")
 
-            task_details = {}
-            task_id = response.get("response").get("taskId")
-
-            while (True):
-                task_details = self.get_task_details(task_id)
-                name = image_name.split('/')[-1]
+                task_details = {}
+                task_id = response.get("response").get("taskId")
 
-                if task_details and \
-                        ("completed successfully" in task_details.get("progress").lower()):
-                    self.result['changed'] = True
-                    self.status = "success"
-                    self.msg = "Swim Image {0} imported successfully".format(name)
-                    self.result['msg'] = self.msg
-                    self.log(self.msg, "INFO")
-                    break
+                # Monitor the task progress
+                while True:
+                    task_details = self.get_task_details(task_id)
+                    if task_details and "completed successfully" in task_details.get("progress", "").lower():
+                        if images_to_import:
+                            images_to_import_str = ", ".join(images_to_import)
+
+                            self.result['changed'] = True
+                            self.status = "success"
+                            self.msg = "Swim Image(s) {0} imported successfully".format(images_to_import_str)
+                            self.result['msg'] = self.msg
+                            self.log(self.msg, "INFO")
+                            break
+
+                    if task_details and task_details.get("isError"):
+                        if "already exists" in task_details.get("failureReason", ""):
+                            self.msg = "SWIM Image {0} already exists in the Cisco Catalyst Center".format(image_name.split('/')[-1])
+                            self.result['msg'] = self.msg
+                            self.log(self.msg, "INFO")
+                            self.status = "success"
+                            self.result['changed'] = False
+                            break
+                        else:
+                            self.status = "failed"
+                            self.msg = task_details.get("failureReason", "SWIM Image {0} seems to be invalid".format(image_name))
+                            self.log(self.msg, "WARNING")
+                            self.result['response'] = self.msg
+                            return self
+
+                self.result['response'] = task_details if task_details else response
```

Comment:
if we don't have task_details, can we return in while loop itself?

Reply:
updated the code